packages/babel-plugin/src/utils/ast.tsx

```diff
@@ -99,6 +130,98 @@ export const getKey = (node: t.Expression) => {
   throw new Error();
 };
 
+/**
+ * Returns `true` if an identifier or any paths that reference the identifier are mutated.
+ * @param path
+ */
+const isIdentifierReferencesMutated = (path: NodePath<t.Identifier>): boolean => {
+  const binding = path.scope.getBinding(path.node.name);
+  if (!binding) {
+    return false;
+  }
+
+  if (!t.isVariableDeclarator(binding.path.node) || !binding.constant) {
```

Comment:
Which case is this one `!t.isVariableDeclarator(binding.path.node)`?

Reply:
this forces all bindings to be

`blah = 'ok';`

variable declarator^

functions declarations could be other things we aren't handling atm. will tackle them when we get to mixins.

Reply:
yeah..thats what I thought of..that we are not handling those cases as of now..!! Just wanted to confirm 😄